service/create/encrypt.go

```diff
@@ -26,53 +26,27 @@ type rawTLSAssets struct {
 	EtcdServerCrt   []byte
 }
 
-// Encrypted PEM encoded TLS assets
-type encryptedTLSAssets struct {
-	APIServerCA     []byte
-	APIServerKey    []byte
-	APIServerCrt    []byte
-	CalicoClientCA  []byte
-	CalicoClientKey []byte
-	CalicoClientCrt []byte
-	EtcdServerCA    []byte
-	EtcdServerKey   []byte
-	EtcdServerCrt   []byte
-}
-
-func readRawTLSAssets(tlsAssetsDir string) (*rawTLSAssets, error) {
-	r := new(rawTLSAssets)
-	files := []struct {
-		name          string
-		ca, key, cert *[]byte
-	}{
-		{"apiserver", &r.APIServerCA, &r.APIServerKey, &r.APIServerCrt},
-		{"calico/client", &r.CalicoClientCA, &r.CalicoClientKey, &r.CalicoClientCrt},
-		{"etcd/server", &r.EtcdServerCA, &r.EtcdServerKey, &r.EtcdServerCrt},
-	}
-	for _, file := range files {
-		caPath := filepath.Join(tlsAssetsDir, file.name+"-ca.pem")
-		keyPath := filepath.Join(tlsAssetsDir, file.name+"-key.pem")
-		certPath := filepath.Join(tlsAssetsDir, file.name+"-crt.pem")
-
-		caData, err := ioutil.ReadFile(caPath)
-		if err != nil {
-			return nil, microerror.MaskAny(err)
-		}
-		*file.ca = caData
+// PEM encoded TLS assets.
+type rawTLSAssets TLSassets
 
-		certData, err := ioutil.ReadFile(certPath)
-		if err != nil {
-			return nil, microerror.MaskAny(err)
-		}
-		*file.cert = certData
+// Encrypted PEM encoded TLS assets
+type encryptedTLSAssets TLSassets
 
-		keyData, err := ioutil.ReadFile(keyPath)
-		if err != nil {
-			return nil, microerror.MaskAny(err)
-		}
-		*file.key = keyData
+func createRawTLSAssets(assets map[string]map[string][]byte) *rawTLSAssets {
+	return &rawTLSAssets{
+		APIServerCA:     assets["apiserver"]["ca"],
```

Comment:
Where do the magic strings come from? Is this a convention? Where are they used as well? Can we try to have one common place to prevent serious issues caused by typos?